.github/workflows/p-token-release.yml

```diff
@@ -0,0 +1,69 @@
+name: "Release P-Token KMIR"
+
+on:
+  push:
+    branches:
+      - feature/p-token
+
+jobs:
+  release-docker:
+    runs-on: [self-hosted, linux, normal]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: 'Setup Docker Buildx'
+        uses: docker/setup-buildx-action@v3.10.0
+
+      - name: 'Login to Docker Hub'
+        uses: docker/login-action@v3.4.0
+        with:
+          username: ${{ vars.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Set Image Name
+        id: set-image-name
+        run: |
+          echo "image-name=runtimeverificationinc/kmir" >> $GITHUB_OUTPUT
+          echo "k-version=$(cat deps/k_release)" >> $GITHUB_OUTPUT
+          echo "short-sha=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Build Kmir Container
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: Dockerfile.kmir
+          platforms: linux/amd64
+          push: true
+          build-args: |
+            K_VERSION=${{ steps.set-image-name.outputs.k-version }}
+          tags: ${{ steps.set-image-name.outputs.image-name }}:p-token-${{ steps.set-image-name.outputs.short-sha }}
```

Comment:
This is the difference to the `release` workflow where we would use the `package/version` string.